source/pictureViewer.cpp

```diff
@@ -275,6 +283,21 @@ void PictureViewer::ResizeToImage() {
    Window()->ResizeBy( - byX, - byY );
 }
 
+void
+PictureViewer::SetZoom(float z)
+{
+	if (z <= 0.0)
+		return;
+
+	imgZoom = z;
```

Comment:
I decided to slightly change that expression to `z >= 0 ? z : imgZoom` because setting the zoom to 0 sets the image to just 1x1 px and I don't think that would be very useful for viewing images. Let me know if I should use the one you gave me or if I should keep it that way.

Reply:
Maybe something like this:
```
		if (z < 0.1 ||  z > 10)
			return;

		imgZoom = z;
```